test/integration/apiserver/flowcontrol/concurrency_test.go

```diff
@@ -312,9 +319,10 @@ func createPriorityLevelAndBindingFlowSchemaForUser(c clientset.Interface, usern
 	})
 }
 
-func streamRequests(parallel int, request func(), stopCh <-chan struct{}) {
+func streamRequests(parallel int, request func(), wg *sync.WaitGroup, stopCh <-chan struct{}) {
 	for i := 0; i < parallel; i++ {
 		go func() {
+			defer wg.Done()
```

Comment:
I'm a little curious here. Seems when we we close `stopCh`, the goroutine will be returned, why we still need the waitgroup here.

Reply:
But we don't want them to just finish - we want them to finish before we actually shutdown apiserver - otherwise the requests start failing with "cannot access apiserver" errors (thus failing test).

Reply:
Got it.

Reply:
> 

got it